smarts/core/sensors.py

```diff
@@ -489,40 +489,43 @@ def _vehicle_is_off_route_and_wrong_way(cls, sim, vehicle):
         if not nearest_lane:
             return (True, False)
 
+        # Check whether vehicle is in wrong-way
+        is_wrong_way = cls._check_wrong_way_event(nearest_lane, vehicle)
+
+        # Check whether vehicle has no-route or is on-route
         if (
-            not route_roads
-            or nearest_lane.road in route_roads
+            not route_roads  # Vehicle has no-route. E.g., endless mission with a random route
+            or nearest_lane.road in route_roads  # Vehicle is on-route
             or nearest_lane.in_junction
         ):
-            is_wrong_way = cls._check_wrong_way_event(nearest_lane, sim, vehicle)
             return (False, is_wrong_way)
 
         veh_offset = nearest_lane.offset_along_lane(vehicle_pos)
 
-        # so we're not obviously on the route, but we might have just gone
+        # so we're obviously not on the route, but we might have just gone
         # over the center line into an oncoming lane...
         for on_lane in nearest_lane.oncoming_lanes_at_offset(veh_offset):
             if on_lane.road in route_roads:
-                is_wrong_way = cls._check_wrong_way_event(on_lane, sim, vehicle)
                 return (False, is_wrong_way)
```

Comment:
Does this case handle when the vehicle is not completely "off route" but part of the vehicle is on the opposing lane? 

So I was wondering if the car goes off the route even **partially** then the wrong way event is triggered (i.e. wrong_way=True), however if I have (wrong_way=False, off_route=True) then will the event trigger will occur when the car is **completely** off the route?